packages/dashboard/src/testing/testing-ground/testing-ground.tsx

```diff
@@ -9,6 +17,42 @@ import { dashboardConfig } from '../mocks';
 export class TestingGround {
   @State() dashboardConfiguration: DashboardConfiguration = dashboardConfig;
 
+  /** Holds all necessary information about dashboard */
+  @State() state: DashboardState = {
+    dashboardConfiguration: dashboardConfig,
+    selectedWidgetIds: [],
+    numTimesCopyGroupHasBeenPasted: 0,
+    copyGroup: [],
+    stretchToFit: DEFAULT_STRETCH_TO_FIT,
+    width: DEFAULT_WIDTH,
+    cellSize: DEFAULT_CELL_SIZE,
+    intermediateDashboardConfiguration: undefined,
+    undoQueue: [],
+    redoQueue: [],
+    previousPosition: undefined,
+  };
+
+  /**
+   * Callback that is fired every time the dashboard configuration has been altered.
+   *
+   * When a widget is moved, resized, deleted, appended, or altered, then this method is called
+   */
+  onDashboardConfigurationChange = (config: DashboardConfiguration) => {
+    this.dashboardConfiguration = config;
+  };
+
+  store: DashboardStore;
+
+  componentWillLoad() {
+    this.state.dashboardConfiguration = this.dashboardConfiguration;
+    this.store = createStore(dashboardReducer, this.state);
+    this.store.subscribe(() => {
+      this.state = this.store.getState();
+      this.state.dashboardConfiguration.widgets = this.state.dashboardConfiguration.widgets.map(trimWidgetPosition);
```

Comment:
unrelated to this your commit, but we should handle the trim widget position instead within the reducer (i don't expect you to fix this in this PR, more of a general note on the codebase)

Reply:
Agreed, it's odd to have it here.